### PR TITLE
ETL: consume dispatch callbacks futures, add tests

### PR DESCRIPTION
**Pull request by kislyuk**

### CODE REVIEW

File: tests/test_etl.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+class TestETLException(Exception):
+    pass
+
+def error_fn(*args, **kwargs):
+    logger.debug(f"Finalizer called with {args} {kwargs}, raising error")
+    raise TestETLException("test")
+
+class TestETL(unittest.TestCase):
+    @unittest.skipIf(sys.version_info < (3, 6), "Only testing under Python 3.6+")
```
Comment (calvinnhieu) [inline tests/test_etl.py]: "Only testing Python 3.6+"?

Comment (calvinnhieu) [reply tests/test_etl.py]: oh I see, I think "under" is just overloaded

File: tests/test_etl.py [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    def test_etl(self):
+        from unittest.mock import MagicMock, patch
+        import dcplib.etl
+
+        files = [
+            {"name": hex(i), "content-type": "application/json", "uuid": hex(i), "version": "1", "sha256": "0"}
+            for i in range(256)
+        ]
```
Comment (calvinnhieu) [inline tests/test_etl.py]: cool

File: tests/test_etl.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+                                        dss_client=MockDSSClient())
+            e.extract(transformer=tf,
+                      loader=ld,
+                      finalizer=fn,
+                      query={"test": True},
+                      max_workers=2,
+                      max_dispatchers=1,
+                      dispatch_executor_class=concurrent.futures.ProcessPoolExecutor)
+            self.assertEqual(calls["tf"], 4)
+            self.assertEqual(calls["ld"], 4)
```
Comment (calvinnhieu) [inline tests/test_etl.py]: `calls` doesn't get incremented if it's run in a separate process?

Comment (kislyuk) [reply tests/test_etl.py]: Correct.

**Pull request merged**